model2vec/train/classifier.py

```diff
@@ -0,0 +1,297 @@
+            return train_test_split(X, y, test_size=test_size, random_state=42, shuffle=True)
+        return train_test_split(X, y, test_size=test_size, random_state=42, shuffle=True, stratify=y)
+
+    def to_pipeline(self) -> StaticModelPipeline:
+        """Convert the model to an sklearn pipeline."""
+        static_model = self.to_static_model()
+
+        random_state = np.random.RandomState(_RANDOM_SEED)
+        n_items = len(self.classes)
+        X = random_state.randn(n_items, static_model.dim)
+        y = self.classes
+
+        converted = make_pipeline(MLPClassifier(hidden_layer_sizes=(self.hidden_dim,) * self.n_layers))
+        converted.fit(X, y)
+        mlp_head: MLPClassifier = converted[-1]
+
+        for index, layer in enumerate([module for module in self.head if isinstance(module, nn.Linear)]):
+            mlp_head.coefs_[index] = layer.weight.detach().cpu().numpy().T
+            mlp_head.intercepts_[index] = layer.bias.detach().cpu().numpy()
+        # Below is necessary to ensure that the converted model works correctly.
+        # In scikit-learn, a binary classifier only has a single vector of output coefficients
+        # and a single intercept. We use two output vectors.
+        # To convert correctly, we need to set the outputs correctly, and fix the activation function.
+        # Make sure n_outputs is set to > 1.
+        mlp_head.n_outputs_ = self.out_dim
+        # Set to softmax
+        mlp_head.out_activation_ = "softmax"
+
+        return StaticModelPipeline(static_model, converted)
+
+
+class _ClassifierLightningModule(pl.LightningModule):
+    def __init__(self, model: StaticModelForClassification, learning_rate: float) -> None:
+        """Initialize the lightningmodule."""
+        super().__init__()
+        self.model = model
+        self.learning_rate = learning_rate
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """Simple forward pass."""
+        return self.model(x)
+
+    def training_step(self, batch: tuple[torch.Tensor, torch.Tensor], batch_idx: int) -> torch.Tensor:
+        """Simple training step using cross entropy loss."""
+        x, y = batch
+        head_out, _ = self.model(x)
+        loss = nn.functional.cross_entropy(head_out, y).mean()
+
+        self.log("train_loss", loss)
+        return loss
+
+    def validation_step(self, batch: tuple[torch.Tensor, torch.Tensor], batch_idx: int) -> torch.Tensor:
+        """Simple validation step using cross entropy loss and accuracy."""
+        x, y = batch
+        head_out, _ = self.model(x)
+        loss = nn.functional.cross_entropy(head_out, y).mean()
+        accuracy = (head_out.argmax(1) == y).float().mean()
+
+        self.log("val_loss", loss)
+        self.log("val_accuracy", accuracy, prog_bar=True)
+
+        return loss
+
+    def configure_optimizers(self) -> OptimizerLRScheduler:
+        """Simple Adam optimizer."""
+        optimizer = torch.optim.Adam(self.model.parameters(), lr=1e-3)
```

Comment:
I think learning rate is now ignored, this should be:
```suggestion
        optimizer = torch.optim.Adam(self.model.parameters(), lr=self.learning_rate)
```